addon/components/frost-select.js

```diff
@@ -364,7 +365,8 @@ export default Component.extend({
       const onInput = this.get('onInput')
 
       if (typeOf(onInput) === 'function') {
-        onInput(filter)
+        const debounceInterval = this.get('debounceInterval')
+        run.debounce(this, onInput, filter, debounceInterval)
```

Comment:
`debounce` will essentially start a setTimeout. If debounced method is fired again, most recent will be set as the one to fire, and the previous call will be cancelled and popped from a queue.

Here's the internal implementation, if you are interested.
https://github.com/ebryn/backburner.js/blob/59cfe851ca0ce9b2e6c7be61435e22fab3731bdb/lib/backburner/index.js#L468